plugin/src/main/resources/META-INF/plugin.xml

```diff
@@ -5,8 +5,7 @@
 
     <xi:include href="/META-INF/fortran-core.xml" xpointer="xpointer(/idea-plugin/*)"/>
 
-    <content>
-        <module name="org.jetbrains.fortran.clion"/>
-        <module name="org.jetbrains.fortran.debugger"/>
-    </content>
+    <depends optional="true" config-file="org.jetbrains.fortran.clion.xml">com.intellij.modules.clion</depends>
+    <depends optional="true" config-file="org.jetbrains.fortran.debugger.xml">com.intellij.modules.clion</depends>
+    <depends optional="true" config-file="org.jetbrains.fortran.nativeDebug.xml">com.intellij.modules.clion</depends>
```

Comment:
So then how does this one work? It is an optional, so it seems to depend on whether or not it is packaged for the specific IDE. But then how do we control if something is packaged for each IDE?

Reply:
Actually I've found the PR related to this https://github.com/intellij-rust/intellij-rust/pull/8709. It looks like they did migrate from one to the other just last year referencing ["plugin model v2"](https://github.com/JetBrains/intellij-community/blob/6f85064d9ff78b6d2290e7cd8ca0a50f12798a3b/docs/plugin.md)

Reply:
I am looking into it once again, thanks for the references.
Maybe you are right and this is a newer way to build plugins. I am not from an IDE team so I am not aware of changes. I thought that `module` is only for IDE parts, not for plugins. I will revert my changes if it was re-designed and create a task to update our docs

Reply:
Looks like you are right and it is a new way of building plugins, it is just unstable and therefore not yet mentioned in the docs. I will rollback my changes

Reply:
Kay, I am also not sure if all the implementation there are correct. I am not a java person so I don't know how to analyze a jar archive or figure out how to check dependencies. But getting the idea debugger to work would give us some clues

Reply:
I've had a discussion with Jakub Chrzanowski on the slack channel. And he is on the impression that `rust` and other plugins should not be using that model yet. However with the `mergeJar` task, it should be working for non-bundled plugins. There is a lot being promised with Gradle 2.0, but the main focus there is on dependencies. But in the meantime, it seems that the new plugin model will not be touched until the actual release of Gradle 2.0, and at that point we would also rework the dependencies here. So until then, we should be safe using this model.